packet/commands.py

```diff
@@ -141,30 +156,64 @@ def ldap_sync():
     print("Fetching data from LDAP...")
     all_upper = {member.uid: member for member in filter(lambda member: not ldap_is_intromember(member),
                                                          ldap_get_active_members())}
+    on_coop = {member.uid: member for member in filter(lambda member: ldap_is_on_coop(member),
+                                                       ldap_get_active_members())}
+
+    rtp = ldap_get_active_rtp()
+    three_da = ldap_get_3da()
+    webmaster = ldap_get_webmaster()
+    c_m = ldap_get_constitutional_maintainer()
+    drink = ldap_get_drink_admin()
 
     print("Applying updates to the DB...")
     for packet in Packet.query.filter(Packet.end > datetime.now()).all():
-        # Update the eboard state of all UpperSignatures
+        # Update the role state of all UpperSignatures
         for sig in filter(lambda sig: sig.member in all_upper, packet.upper_signatures):
```

Comment:
I'd say it's worth checking for co-ops when updating from ldap as well, since that can change over the course of a packet season.

Reply:
So that is handled by the existing logic. If someone is on co-op and they're in the active group, they're treated as though they're alumni signatures, and if someone was active and enrolled in co-op late, they're converted at a later stage

Reply:
Oh I missed the second move to Misc for loop, sorry,